src/coreclr/src/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.Windows.cs

```diff
@@ -19,43 +16,46 @@ internal static partial class GlobalizationMode
 
         private static bool LoadIcu()
         {
-            if (!TryGetAppLocalIcuSwitchValue(out ReadOnlySpan<char> version, out ReadOnlySpan<char> suffix))
+            if (!TryGetAppLocalIcuSwitchValue(out string? icuSuffixAndVersion))
             {
                 return Interop.Globalization.LoadICU() != 0;
             }
 
-            string extension = ".dll";
-            string icuucBase = "icuuc";
-            string icuinBase = "icuin";
+            LoadAppLocalIcu(icuSuffixAndVersion);
+            return true;
+        }
+
+        private static void LoadAppLocalIcuCore(ReadOnlySpan<char> version, ReadOnlySpan<char> suffix)
+        {
+            const string extension = ".dll";
+            const string icuucBase = "icuuc";
+            const string icuinBase = "icuin";
             IntPtr icuucLib = IntPtr.Zero;
             IntPtr icuinLib = IntPtr.Zero;
-            Assembly assembly = Assembly.GetExecutingAssembly();
-
-            int index = version.ToString().IndexOf('.', StringComparison.Ordinal);
 
-            if (index != -1)
+            int index = version.IndexOf('.');
+            if (index > 0)
```

Comment:
nit: >=

Reply:
The reason why I did `> 0` is because if `.` is in position 0, it means we got `.` or `.65` or something like that, so it is not worth trying to load icu with `.` first and then with `.65` as it is not a valid version, right?  It doesn't matter either way works.

Reply:
Isn't equivalent to no major version is provided?

Reply:
Right, but is there even a way to have ICU with no major version?

Reply:
The builder can decide that. no? if you think we shouldn't consider it, then that is ok. we want to ensure documenting the behavior if we run into such a case.

Reply:
I can’t even think how the name would be there for the binaries. In that case I would think the user should pass down 0.54. In windows unless the user changes how ICU is built it encodes the major version in the binary only at the moment.

Reply:
I think we should just document it as I don’t even think it is going to be that common of a scenario.